packages/core/src/types.ts

```diff
@@ -230,10 +230,13 @@ export type Receiver<TEvent extends EventObject> = (
   listener: (event: TEvent) => void
 ) => void;
 
-export type InvokeCallback<TEvent extends EventObject = AnyEventObject> = (
+export type InvokeCallback<
+  TEvent extends EventObject = AnyEventObject,
+  TReceivedEvent extends EventObject = AnyEventObject
+> = (
```

Comment:
For most types, we use `TEvent` as a type for events that the actor **receives** and not for events that it sends. So it kinda feels like the naming here should be flipped - `TReceivedEvent` should become `TEvent` and `TEvent` should become `TSentEvent`. There is already a precedence for `TSentEvent` here: https://github.com/davidkpiano/xstate/blob/18821577bd7572a00eecaf563c098526db06c854/packages/core/src/actions.ts#L196

Reply:
1e883f44cf6d3cddb823944da847f793121e13fc